source/fab/util.py

```diff
@@ -62,6 +62,31 @@ def file_checksum(fpath):
         return HashedFile(fpath, zlib.crc32(infile.read()))
 
 
+def remove_minus_J(flags, verbose=False):
+    """
+    Remove the -J <folder> from the given flags.
+
+    """
+    # todo: what if there's two -Js ?
+    # todo: Fab should be compiler aware, with the possibly of different flags to ignore per compiler
```

Comment:
As mentioned elsewhere, I think that ideally we wouldn't have to pick apart already constructed argument lists.